pkg/alertmanager/operator.go

```diff
@@ -654,6 +658,19 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 
 	level.Info(logger).Log("msg", "sync alertmanager")
 
+	// Check if operator has the permission to read the storage class.
+	if c.canReadStorageClass {
+		// Check if the AlertManager Specs is using a Volume Claim.
+		if !reflect.DeepEqual(am.Spec.Storage.VolumeClaimTemplate, monitoringv1.EmbeddedPersistentVolumeClaim{}) {
```

Comment:
can't we do this instead?
```suggestion
		if am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName != nil {
```

Reply:
Sure, will update the code